voicevox_engine/guided/extractor.py

```diff
@@ -0,0 +1,104 @@
+import os
+from typing import List, Optional
+
+import numpy as np
+import pyworld as pw
+import snfa
+from scipy.signal import resample
+from snfa.viterbi import Segment
+
+from ..model import AudioQuery
+
+aligner: Optional[snfa.Aligner] = None
+
+
+def _lazy_init(model_path: str):
+    global aligner
+    if aligner is None:
+        if not os.path.exists(model_path):
+            raise Exception("No model exists, check the build pipeline")
+        aligner = snfa.Aligner(model_path)
+
+
+def _query2phoneme(query: AudioQuery) -> List[str]:
+    phoneme_list = []
+    for accent_phrase in query.accent_phrases:
+        for mora in accent_phrase.moras:
+            if mora.consonant is not None:
+                phoneme_list.append(mora.consonant)
+            phoneme_list.append(mora.vowel)
+        if accent_phrase.pause_mora:
+            phoneme_list.append("pau")
+    return ["pau"] + phoneme_list + ["pau"]
+
+
+def _align(wav: np.ndarray, src_sr: int, query: AudioQuery):
+    if len(wav.shape) == 2:
+        wav = np.sum(wav, axis=1) / 2
+    # convert to int, to avoid overflow in np.int32 type
+    wav = resample(wav, int(aligner.sr) * int(wav.shape[0]) // src_sr)
```

Comment:
This might be a better approach?
```suggestion
    # convert to int, to avoid overflow in np.int32 type
    wav = resample(wav, int(aligner.sr * wav.shape[0]) // src_sr)
```